lib/net/ssh/service/forward.rb

```diff
@@ -120,6 +121,59 @@ def active_locals
       @local_forwarded_ports.keys
     end
 
+    # Starts listening for connections on the local host, and forwards them
+    # to the specified remote socket via the SSH connection. This will
+    # (re)create the local socket file. The remote server needs to have the
+    # socket file already available.
+    #
+    #   ssh.forward.local_socket('/tmp/local.sock', '/tmp/remote.sock')
+    def local_socket(local_socket_path, remote_socket_path)
+      File.delete(local_socket_path) if File.exist?(local_socket_path)
+      socket = Socket.unix_server_socket(local_socket_path)
+
+      @local_forwarded_sockets[local_socket_path] = socket
+
+      session.listen_to(socket) do |server|
+        client = server.accept[0]
+        debug { "received connection on #{socket}" }
+
+        channel = session.open_channel("direct-streamlocal@openssh.com",
+                                       :string, remote_socket_path,
+                                       :string, nil,
+                                       :long, 0) do |achannel|
+          achannel.info { "direct channel established" }
+        end
+
+        prepare_client(client, channel, :local)
+
+        channel.on_open_failed do |ch, code, description|
+          channel.error { "could not establish direct channel: #{description} (#{code})" }
+          session.stop_listening_to(channel[:socket])
+          channel[:socket].close
+        end
+      end
+
+      local_socket_path
+    end
+
+    # Terminates an active local forwarded socket. If no such forwarded socket
+    # exists, this will raise an exception. Otherwise, the forwarded connection
```

Comment:
i don't think it will raise an exception, will it? I'm fine both ways raising or not, but docs should state what the code is doing.